From 104727ad6ea255d2f774c3140ccadde68a7265fb Mon Sep 17 00:00:00 2001
From: Milan Broz <mbroz@redhat.com>
Date: Thu, 23 Aug 2012 19:55:27 +0200
Subject: [PATCH] Require fipscheck and libssl in FIPS module

To properly perform verification in FIPS mode,
we need to install fipscheck and libssl explicitly.

(cryptsetup seems to be the first user of this verification in ramdisk...)

Signed-off-by: Milan Broz <mbroz@redhat.com>
---
 modules.d/01fips/module-setup.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/01fips/module-setup.sh b/modules.d/01fips/module-setup.sh
index 2517964..103a479 100755
--- a/modules.d/01fips/module-setup.sh
+++ b/modules.d/01fips/module-setup.sh
@@ -33,11 +33,11 @@ install() {
     inst_hook pre-pivot 01 "$moddir/fips-noboot.sh"
     inst_script "$moddir/fips.sh" /sbin/fips.sh
 
-    dracut_install sha512hmac rmmod insmod mount uname umount
+    dracut_install sha512hmac rmmod insmod mount uname umount fipscheck
 
     inst_libdir_file libsoftokn3.so libsoftokn3.so \
         libsoftokn3.chk libfreebl3.so libfreebl3.chk \
-        'hmaccalc/sha512hmac.hmac'
+        libssl.so 'hmaccalc/sha512hmac.hmac'
 
     dracut_install -o prelink
 }
